### DISCUSSION TITLE
missing parts or conscious choice

### DISCUSSION THREAD

**Discussion by Heelderpeel** [astrandb/miele#97] [Q&A] [answered]
Hi, I've been playing around with the old and your Miele integration, but I'm missing a number of entities in both my washing machine and dryer that I did have in the old version. (example door open and the energy counter) is that a conscious choice or not. Or should I consider that a bug?

**Comment by astrandb** [comment, marked as answer]
The short answer is conscious choice. The basic design idea was to create sensors and other entities according to the API capabilities matrix from Miele (v1.0.5). However, this matrix is incorrect in many places. Some devices don't support a certain feature, and some devices support features even though they are marked as unsupported.
I can easily turn on the door sensor for washing machines and dryers.

The energy sensors should be working. They are disabled by default. You can unhide/activate them on the device page.

**Reply by Heelderpeel** [reply]
Hello @astrandb
I see you already changed it, thank you very much for the fast service.
that's very nice.
Energy consumption is included indeed for the washing machine, but not for the tumbledryer.
Is there a possibility to add that?

**Reply by astrandb** [reply]
👍

**Comment by Heelderpeel** [comment]
Hello, only with the tumble dryer I can not turn on the energy consumption, it is not included.

**Reply by astrandb** [reply]
Fixed in PR #105

**Reply by Heelderpeel** [reply]
very nice thnx
